src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -27,36 +27,57 @@ class CreateZuoraSubscription(servicesProvider: ServiceProvider = ServiceProvide
     requestInfo: RequestInfo,
     context: Context,
     services: Services
-  ): FutureHandlerResult = for {
-    identityId <- Future.fromTry(IdentityId(state.user.id))
-    maybeDomainSubscription <- GetRecurringSubscription(services.zuoraService, state.requestId, identityId, state.product.billingPeriod)
-    thankYouState <- maybeDomainSubscription match {
-      case Some(domainSubscription) => skipSubscribe(state, requestInfo, domainSubscription)
-      case None => subscribe(state, requestInfo, services)
-    }
-  } yield thankYouState
+  ): FutureHandlerResult = {
+    val subscribeItem = buildSubscribeItem(state, services.promotionService)
+    for {
+      identityId <- Future.fromTry(IdentityId(state.user.id))
+      maybeDomainSubscription <- GetRecurringSubscription(services.zuoraService, state.requestId, identityId, state.product.billingPeriod)
+      previewPaymentSchedule <- PreviewPaymentSchedule(subscribeItem, state.product.billingPeriod, services, checkSingleResponse)
```

Comment:
The preview call is made in the CreateZuoraSubscription lambda. I've put this here because we could use the result to perform safety checks before actually setting up subscriptions in the future.

Reply:
Good idea 👍